src/tls13.c

```diff
@@ -4437,7 +4437,8 @@ static int RestartHandshakeHashWithCookie(WOLFSSL* ssl, Cookie* cookie)
     length = HRR_BODY_SZ - ID_LEN + ssl->session->sessionIDSz +
              HRR_COOKIE_HDR_SZ + cookie->len;
     length += HRR_VERSIONS_SZ;
-    if (cookieDataSz > hashSz + OPAQUE16_LEN) {
+    /* HashSz (1 byte) + Hash (HashSz bytes) + CipherSuite (2 bytes) */
+    if (cookieDataSz > hashSz + OPAQUE16_LEN + OPAQUE8_LEN) {
```

Comment:
I like to have the elements added in the order that they appear in memory ie: tag + length value + length. I think it makes it easier to read, especially in more complicated encoding.